src/Data/List/Base.agda

```diff
@@ -228,130 +223,117 @@ unfold B f {n = suc n} s with f s
 ------------------------------------------------------------------------
 -- Operations for deconstructing lists
 
--- Note that although these combinators can be useful for programming, when
--- proving it is often a better idea to manually destruct a list argument:
--- each branch of the pattern-matching will have a refined type.
+-- Note that although the following three combinators can be useful for
+-- programming, when proving it is often a better idea to manually
+-- destruct a list argument as each branch of the pattern-matching will
+-- have a refined type.
 
-module _ {a} {A : Set a} where
+uncons : List A → Maybe (A × List A)
+uncons []       = nothing
+uncons (x ∷ xs) = just (x , xs)
 
-  uncons : List A → Maybe (A × List A)
-  uncons []       = nothing
-  uncons (x ∷ xs) = just (x , xs)
+head : List A → Maybe A
+head []      = nothing
+head (x ∷ _) = just x
 
-  head : List A → Maybe A
-  head []      = nothing
-  head (x ∷ _) = just x
+tail : List A → Maybe (List A)
+tail []       = nothing
+tail (_ ∷ xs) = just xs
 
-  tail : List A → Maybe (List A)
-  tail []       = nothing
-  tail (_ ∷ xs) = just xs
-
-take : ∀ {a} {A : Set a} → ℕ → List A → List A
+take : ℕ → List A → List A
 take zero    xs       = []
 take (suc n) []       = []
 take (suc n) (x ∷ xs) = x ∷ take n xs
 
-drop : ∀ {a} {A : Set a} → ℕ → List A → List A
+drop : ℕ → List A → List A
 drop zero    xs       = xs
 drop (suc n) []       = []
 drop (suc n) (x ∷ xs) = drop n xs
 
-splitAt : ∀ {a} {A : Set a} → ℕ → List A → (List A × List A)
+splitAt : ℕ → List A → (List A × List A)
 splitAt zero    xs       = ([] , xs)
 splitAt (suc n) []       = ([] , [])
 splitAt (suc n) (x ∷ xs) with splitAt n xs
 ... | (ys , zs) = (x ∷ ys , zs)
 
-takeWhile : ∀ {a p} {A : Set a} {P : Pred A p} →
-            Decidable P → List A → List A
+takeWhile : ∀ {p} {P : Pred A p} → Decidable P → List A → List A
 takeWhile P? []       = []
```

Comment:
You can drop `{p}` for all of these (or should remove `p` from the list of Level variables).